x-pack/legacy/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -48,6 +48,24 @@ function isLocalStateDirty(
   );
 }
 
+// Exported for testing purposes, sets the Kibana breadcrumbs.
+export function setBreadcrumbs(
+  chrome: {
+    addBasePath: (url: string) => string;
+    breadcrumbs: { set: (crumbs: Array<{ text: string; href?: string }>) => void };
+  },
+  persistedDoc?: { title: string }
+) {
+  const rootCrumb = {
+    href: chrome.addBasePath(`/app/kibana#/visualize`),
```

Comment:
Not sure whether we want to open that can of worms, but you can navigate back to the visualize listing with unsaved changes and lose them because we don't persist to the url. 

Maps solves this by a browser prompt, either that or a modal would be nice
<img width="542" alt="Screenshot 2019-08-30 at 09 38 33" src="https://user-images.githubusercontent.com/1508364/64002297-feb61200-cb09-11e9-99f2-8efab058b1ee.png">

The same thing would apply to navigating away from the page by clicking another app or hitting back, so probably out of scope for this PR.

Reply:
Agreed. I think it's unrelated, though. I'll make sure we have an issue for warning when you're about to lose changes.